sentry_dsn_data.go

```diff
@@ -0,0 +1,163 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/getsentry/sentry-go"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var DSN = "dsn"
+
+// map from Sentry DSN to Client
+type DsnData struct {
+	mutex     sync.RWMutex
+	clientMap map[string]*sentry.Client
+}
+
+func NewDsnData() *DsnData {
+	return &DsnData{
+		mutex:     sync.RWMutex{},
+		clientMap: make(map[string]*sentry.Client),
+	}
+}
+
+// return client if added successfully
+func (d *DsnData) AddClient(options sentry.ClientOptions) (*sentry.Client, error) {
+	d.mutex.Lock()
+	defer d.mutex.Unlock()
+
+	// check if we already encountered this dsn
+	existingClient, ok := d.clientMap[options.Dsn]
+	if ok {
+		return existingClient, errors.New("a client with the given dsn already exists")
```

Comment:
It's somewhat confusing to return a valid client but also an error. 
Let's not do the "if-exists" check at all, if funciton users care about that, they can call `GetClient`.